feincms3_sites/middleware.py

```diff
@@ -23,10 +25,9 @@ def current_site():
 
 
 def site_middleware(get_response):
-    from .models import Site
-
     def middleware(request):
-        request.site = Site.objects.for_host(request.get_host())
+        site_model = get_site_model()
```

Comment:
Django seems to fetch the user model when importing modules, see e.g. https://github.com/django/django/blob/c5776bfca9e3f35e0ab5aacbdc1a4dbfe92fdfd1/django/contrib/auth/backends.py#L5

Would that be possible here as well?

Reply:
This does not work as the queryset derived manager from `AbstractSite` is bound to the model class that it was registered with, reverting this back gives me the site model was swapped for error and that the manager cannot be used with that.

Reply:
Makes sense.